src/main/java/liquibase/ext/clickhouse/sqlgenerator/CreateDatabaseChangeLogLockTableClickHouse.java

```diff
@@ -40,20 +43,46 @@ public boolean supports(CreateDatabaseChangeLogLockTableStatement statement, Dat
     return database instanceof ClickHouseDatabase;
   }
 
+  private String generateSqlTemplate(boolean replicated) {
+    StringBuilder query = new StringBuilder();
+    query.append("CREATE TABLE IF NOT EXISTS %s.%s ");
+
+    if (replicated) query.append("ON CLUSTER '%s' ");
+
+    query.append("(");
+    query.append("ID Int64,");
+    query.append("LOCKED UInt8,");
+    query.append("LOCKGRANTED Nullable(DateTime64),");
+    query.append("LOCKEDBY Nullable(String)) ");
+
+    if (replicated) query.append("ENGINE ReplicatedMergeTree('%s','%s') ORDER BY ID");
+    else query.append("ENGINE MergeTree() ORDER BY ID");
+
+    return query.toString();
+  }
+
   @Override
   public Sql[] generateSql(
       CreateDatabaseChangeLogLockTableStatement statement,
       Database database,
       SqlGeneratorChain sqlGeneratorChain) {
+    Map<String, String> properties = ParamsLoader.getLiquibaseClickhouseProperties();
+    boolean isReplicated = properties != null;
+
     String createTableQuery =
-        String.format(
-            "CREATE TABLE IF NOT EXISTS %s.%s ("
-                + "ID Int64,"
-                + "LOCKED UInt8,"
-                + "LOCKGRANTED Nullable(DateTime64),"
-                + "LOCKEDBY Nullable(String)) "
-                + "ENGINE MergeTree() ORDER BY ID",
-            database.getDefaultSchemaName(), database.getDatabaseChangeLogLockTableName());
+        !isReplicated
```

Comment:
Style rule: prefer to avoid negations in these simple cases because it is easier to read